modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/Utils.java

```diff
@@ -236,6 +235,25 @@ public class Utils {
         MOSAIC_SUPPORT_FILES_FILTER = initMosaicSupportFilesFilter();
     }
 
+    /** Specific classes to allow when deserializing SampleImage objects */
+    private static final Class<?>[] SAMPLE_IMAGE_CLASSES = {
+        SampleImage.class,
+        Number.class,
+        String.class,
+        byte[].class,
+        float[].class,
+        int[].class,
+        short[].class,
+        java.math.BigInteger.class,
+        java.net.InetAddress.class,
+        java.util.Hashtable.class
+    };
+
+    /** Patterns for qualified class names to allow when deserializing SampleImage objects */
+    private static final String[] SAMPLE_IMAGE_PATTERNS = {
```

Comment:
Yep, you stumbled into the reason why serialization was used in the first place. There are so many possible combinations of sample and color models that it's not practical to try and enumerate them all.

This has some potential to break existing implementations that use custom-written image readers. The patterns here should be made configurable with a system variable.

Reply:
I added a property to allow administrators to allow additional classes for sample image files.

Reply:
Thanks, that helps. It should be also noted in the upgrade notes (both the stricter checks and the system variable to allow other classes be deserialized).